### PR TITLE
Update docs for AzureML deployment plugin

### PR DESCRIPTION
**Pull request by trangevi**
## What changes are proposed in this pull request?

Adding a docs link to point to the new AzureML deployment plugin, and marking the old AzureML handling as deprecated

## How is this patch tested?

Docs change, no tests

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

`mlflow.azureml` package is deprecated in favor of the new AzureML deployment plugin.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [x] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/azureml/__init__.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def build_image(
         return image, registered_model
 
 
-@experimental
+@deprecated("the azureml deployment plugin", since="1.19.0")
```
Comment (dbczumar) [inline mlflow/azureml/__init__.py:240]: Can we include a plugin URL in the message?

File: mlflow/azureml/cli.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def commands():
         " that include the model path, the model run id (if specified), and more."
     ),
 )
-@experimental
+@deprecated(since="1.19.0")
```
Comment (dbczumar) [inline mlflow/azureml/cli.py:82]: Can we add the `alternative` parameter here, as we did for the `deploy()` API?

File: mlflow/azureml/__init__.py [hunk 3]
```diff
@@ -28,7 +28,7 @@
 _logger = logging.getLogger(__name__)
 
 
-@experimental
+@deprecated(since="1.19.0")
```
Comment (dbczumar) [inline mlflow/azureml/__init__.py:31]: Can we add the `alternative` parameter here, as we did for the `deploy()` API?

**Review by dbczumar — COMMENTED** [review]
@trangevi Thanks a bunch for filing this! Just a couple small requests for additions to the deprecation notice.

File: mlflow/azureml/__init__.py (hunk 3 above)
Comment (dbczumar) [reply mlflow/azureml/__init__.py:31]: @trangevi `build_image()` was used as a step in the AzureML deployment procedure, so it makes sense to direct users towards the new deployment procedure, even if the new procedure doesn't involve a separate image building step. Can we specify the deployment plugin as an alternative?

File: mlflow/azureml/cli.py (hunk 2 above)
Comment (trangevi) [reply mlflow/azureml/cli.py:82]: There is no alternative for this, we've completely deprecated it on our end.

File: mlflow/azureml/__init__.py (hunk 1 above)
Comment (trangevi) [reply mlflow/azureml/__init__.py]: Updated with a short url to our docs

File: mlflow/azureml/cli.py (hunk 2 above)
Comment (dbczumar) [reply mlflow/azureml/cli.py:82]: @trangevi `build_image()` was used as a step in the AzureML deployment procedure, so it makes sense to direct users towards the new deployment procedure, even if the new procedure doesn't involve a separate image building step. Can we specify the deployment plugin as an alternative?

Comment (trangevi) [reply mlflow/azureml/cli.py]: Done

File: mlflow/azureml/__init__.py (hunk 3 above)
Comment (trangevi) [reply mlflow/azureml/__init__.py]: Done

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @trangevi !

**Pull request merged**